### PR TITLE
[DigitalRiver] Add DigitalRiver Gateway - store and purchase

### PR DESCRIPTION
**Pull request by pohor**
This PR adds support for DigitalRiver - store and purchase actions.

It uses the `digital_river` gem which is a wrapper around DigitalRiver API created by Chargify.

- The `store` action creates a Customer resource if one does not already exist and attaches a Source to that Customer.

- Th `purchase` action checkes the order state and returns a failure if the order is not in `accepted` state. If it is, it creates a `fulfillment` and returns the `capture id` for the order. For now we assume that there will only be one charge per one order.

The remote tests cannot test the first of the purchase paths due to the lack of a test card that would provide such a scenario.
The unit test covers it instead.

### CODE REVIEW

**Comment by pohor** [comment]
Screenshot 2021-04-27 at 09 28 51

File: test/remote/gateways/remote_digital_river_test.rb [hunk 1]
```diff
@@ -0,0 +1,169 @@
+require 'test_helper'
+require 'digital_river'
+
+class RemoteDigitalRiverTest < Test::Unit::TestCase
+  def setup
+    @gateway = DigitalRiverGateway.new(fixtures(:digital_river))
+    @digital_river_backend = @gateway.instance_eval{@digital_river_gateway}
```
Comment (dbackowski) [inline test/remote/gateways/remote_digital_river_test.rb:7]: I think that using the `instance_variable_get` method would be a better idea here:

```ruby
@gateway.instance_variable_get(:@digital_river_gateway)
```

**Comment by pohor** [comment]
Screenshot 2021-04-27 at 11 49 58

**Review by dbackowski — APPROVED** [review]
Looks good! 👏 🚀

**Review by przemyslawhaptas — APPROVED** [review]
LGTM! 👌

**Pull request merged**